numpy/core/src/umath/ufunc_object.c

```diff
@@ -3793,22 +3787,14 @@ PyUFunc_GenericReduction(PyUFuncObject *ufunc, PyObject *args,
         if (naxes < 0 || naxes > NPY_MAXDIMS) {
             PyErr_SetString(PyExc_ValueError,
                     "too many values for 'axis'");
-            Py_XDECREF(otype);
-            Py_DECREF(mp);
-            return NULL;
+            goto fail;
         }
         for (i = 0; i < naxes; ++i) {
             PyObject *tmp = PyTuple_GET_ITEM(axes_in, i);
             int axis = PyArray_PyIntAsInt(tmp);
-            if (error_converting(axis)) {
-                Py_XDECREF(otype);
-                Py_DECREF(mp);
-                return NULL;
-            }
-            if (check_and_adjust_axis(&axis, ndim) < 0) {
-                Py_XDECREF(otype);
-                Py_DECREF(mp);
-                return NULL;
+            if (error_converting(axis) ||
+                (check_and_adjust_axis(&axis, ndim) < 0)) {
+                goto fail;
```

Comment:
I think this would be clearer as two separate `if`s, like it was before - it makes it look more like the other uses of `check_and_adjust_axis`.